src/filewatch.erl

```diff
@@ -1,107 +1,134 @@
 -module(filewatch).
--export([start/2, stop/1]).
+-export([start/1,
+         start/2,
+         stop/1]).
+
+-type watch_pair() :: {file:name_all(), term()}.
+-export_type([watch_pair/0]).
+
+-define(DRIVER_NAME, filewatch).
+-define(COOLDOWN_MS, application:get_env(filewatch, cooldown_ms, 3000)).
+-define(CALL_ARG, 1).
 
 -type handle() :: pid().
+-type watch_descriptor() :: integer().
+-type directory() :: file:filename().
 
 -record(state,
         {pid :: pid(),
          port :: port(),
-         map :: #{ {integer(), file:filename()} => any(),
-                   file:filename() => integer()}
+         watch_map :: #{
+           {watch_descriptor(), file:filename()} => any(),
+            directory()                          => watch_descriptor()
```

Comment:
That's right. I stuck with using one map because I thought it'd be easier, and because I was certain there'd be no collisions between the different types of keys. I could split it out into two if you prefer.